apps/studio/components/grid/components/editor/JsonEditor.tsx

```diff
@@ -36,6 +36,9 @@ export const JsonEditor = <TRow, TSummaryRow = unknown>({
   }, [])
 
   const saveChanges = useCallback((newValue: string | null) => {
+    if (newValue && newValue.slice(-3, -2) === ',') {
```

Comment:
We should not only check for the character on the last element, we should remove trailing commas in all places - we can probably skip the check and just check for `newValue` to be truthy and apply the regex replace. This will ensure _all_ occurrences of trailing commas are properly replaced

We also need to ensure that we don't replace false positives, some unit tests for that would be great, as any false positives/wrong replacements would be very destructive

Additionally, a comment would be great, as it's hard to just parse a regex on the fly without some context why/what's happening

Reply:
I'm not sure I full understand what you mean by trailing commas in all places. Currently, the JSON editor does not except an entry/value which contains a comma after the last item in the object. For example:

```
{
   "a": "one",
   "b": "two"**,**
}
```

The JSON editor does not accept the above entry as there is comma after `"b": "two"`. The purpose of this PR is to detect if there is a comma after the last item in the object (which is why the `newValue.slice(-3, -2) === ','` check is performed) and replaces it using regex.

If you could elaborate a little bit more on your suggestions, I would love to make the changes!

Reply:
A valid JSON is also an array with multiple objects, i.e.

```
[
{
   "fruit1": "apple",
   "fruit2": "banana",
},
{
   "fruit1": "apple",
   "fruit2": "banana"
}
]
```

If the trailing comma is on another element (not just the last one), this should also be recognized and fixed

Reply:
@kevcodez Thanks for the explanation. I've updated my code accordingly. 

The regular expression now searches for any comma that immediately precedes a closing curly brace, "}" or a closing square bracket, "]", which indicates that it is a trailing comma. I also noticed that the SideEditor JSON Editor returns is also not able to handle trailing commas, so I utilized the same regular expression to check for, and remove any trailing commas when saving on that editor. 

I also added comments to both of the editor files to explain the logic.